src/main/java/commands/ShowRateCommand.java

```diff
@@ -0,0 +1,86 @@
+package commands;
+
+import access.Access;
+import manager.card.Card;
+import manager.chapter.Chapter;
+import storage.Storage;
+import ui.Ui;
+
+import java.util.ArrayList;
+
+public class ShowRateCommand extends Command {
+    public static final String COMMAND_WORD = "showrate";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Show rate percentage in chapter based on individual cards. \n"
+            + "Only available at chapter level. \n"
+            + "Example: " + COMMAND_WORD + "\n";
+
+    public static final String MESSAGE_NO_CARDS_IN_CHAPTER = "You currently have no cards in %1$s";
+    public static final String MESSAGE_SHOW_REVISE_PROMPT = "Are you sure you want to revise this? (Y/N)";
+    public static final String MESSAGE_SHOW_EASY_PROMPT = "The percentage of card that is labeled <easy> is: %1$.2f";
+    public static final String MESSAGE_SHOW_MEDIUM_PROMPT = "The percentage of card that is "
+            + "labeled <medium> is: %1$.2f";
+    public static final String MESSAGE_SHOW_HARD_PROMPT = "The percentage of card that is labeled <hard> is: %1$.2f";
+    public static final String MESSAGE_SHOW_CANNOT_ANSWER_PROMPT = "The percentage of "
+            + "card that is labeled <cannot answer> is: %1$.2f";
+
+    private Double easyPercentage;
+    private Double mediumPercentage;
+    private Double hardPercentage;
+    private Double cannotAnswerPercentage;
+    private int easyCard = 0;
+    private int mediumCard = 0;
+    private int hardCard = 0;
+    private int cannotAnswerCard = 0;
+
+
+    @Override
+    public void execute(Ui ui, Access access, Storage storage) {
+        Chapter chapter = access.getChapter();
+        computePercentage(chapter, ui);
+
+    }
+
+    private void computePercentage(Chapter chapter, Ui ui) {
+        ArrayList<Card> allCards = chapter.getCards().getAllCards();
+        int cardCount = allCards.size();
+        ui.showToUser("\nCard count: " + cardCount);
+        if (cardCount == 0) {
+            ui.showToUser(String.format(MESSAGE_NO_CARDS_IN_CHAPTER, chapter));
```

Comment:
Perhaps could return this to execute and do ui.showToUser there

Reply:
I have modified the method to return boolean instead of String because the execute method will need to print several messages for standard situation, therefore, I use boolean for convenience.